### DISCUSSION TITLE
Unable to find spectator

### DISCUSSION THREAD

**Discussion by bwspsu** [carla-simulator/carla#4782] [Q&A]
Hi everyone, 
             I was trying to run the example code vehicle_gallery.py in PythonAPI/examples using CARLA 0.9.12 built from source, on ubuntu 18.04.
              And I met this error:  unable to find spectator
              This error was raised by the line: spectator = world.get_spectator()
              And the map was the default one (Town10)
              I could not find a solution online.  Does any one know how to solve this? Any help is appreciated !
![errorScreenShot](https://user-images.githubusercontent.com/65965783/138572919-087bc993-a831-4520-8c0e-c75cb46d88db.png)

**Comment by MattRowe18** [comment]
Hi 

Could you give us some context about how you are running CARLA to help us reproduce the problem:

1. Are you running it on your workstation or on a server? 
2. When you launch CARLA do you see a window open up with an aerial view of Town10?
3. Are you launching CARLA with any command line arguments or custom options?

Thank you

**Reply by bwspsu** [reply]
Hi Matt,
             My sincere apologies for the delay of reply!
             The answers to your questions are:
              1.  I'm running it on my personal computer with a ubuntu OS.
              2. when launch CARLA, I can see the view of Town10 in UE4 editor, see the figure below. 
              ![Screenshot from 2021-10-29 02-45-58](https://user-images.githubusercontent.com/65965783/139388542-0d0dd9dd-1300-4670-986d-7f627748c8e5.png)
              3. I run the command "make launch" without any other custom options to launch CARLA.

Other information: When I started to play the simulation in UE4, I saw an error "can not find spectator" in the log in the terminal where I run "make launch", see figure below. 
![Screenshot from 2021-10-29 02-42-45](https://user-images.githubusercontent.com/65965783/139388844-0c1a418e-0b16-4721-8406-60493f9a7233.png)


Thanks a lot in advance!

**Reply by MattRowe18** [reply]
Hi 

Thanks for the answer and the screenshot, this makes things clearer. We think that the problem is the mode you are running UE4 editor. The play button has multiple modes, one is without spectator, the other is with the spectator. It seems from the screenshot you sent that you are running UE4 in the mode without spectator. Try using the other mode of the play button, hopefully that should fix the problem.

**Reply by bwspsu** [reply]
Hi Matt,
            Yes this fixed the problem,  thanks a lot for your help!
            By the way I have another question about importing a customized map, should I continue to post it here, or do I need to start another discussion then post there so that it's easier for others to follow if they have a similar question ?
